submission-receipt/main.py

```diff
@@ -0,0 +1,103 @@
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  Authors:
+    Edmund Su
+"""
+
+import argparse
+import requests
+import json
+import csv
+import sys
+import json
+
+def main():
+    """
+    Python implementation of tool: sanity-check
+    This is auto-generated Python code, please update as needed!
+    """
+
+    parser = argparse.ArgumentParser(description='Tool: sanity-check')
+    parser.add_argument('-s', '--study_id', dest='study_id',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    parser.add_argument('-a', '--analysis_id', dest='analysis_id',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    parser.add_argument('-u', '--submission_song_url', dest='submission_song_url',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    parser.add_argument('-f', '--files', dest='files',nargs='+',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    parser.add_argument('-o', '--output_file', dest='output_file',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    args = parser.parse_args()
+
+    header={"accept":"*/*"}
+    url=args.submission_song_url+"/studies/%s/analysis/%s" % (args.study_id,args.analysis_id)
+    try:
+        response=requests.get(url,headers=header)
+    except requests.exceptions.RequestException as e:  # This is the correct syntax
+        print("Unable to establish connection")
+        raise SystemExit(e)
+
+    if response.status_code==200:
+        with open(args.output_file, 'w', newline='') as csvfile:
+            metadata={
+                "analysisState":None,
+                'publishedAt':None,
+                'submitterSampleId':None,
+                'submitterSpecimenId':None,
+                'submitterDonorId':None,
+                'sampleId':None,
+                'specimenId':None,
+                "donorId":None,
+                "analysisId":None,
+                "studyId":None,
+                "objectId":None,
+                "fileName":None,
+                "fileMd5sum":None
+                }
+    
+            fieldnames = metadata.keys()
+            writer = csv.DictWriter(csvfile, fieldnames=fieldnames,delimiter='\t')
+            writer.writeheader()
+            for file in response.json()['files']:
+                if file['fileName'] not in args.files:
+                    sys.exit("Specified file %s was not found in analysis %s" % (file['fileName'],args.analysis_id))
```

Comment:
This will report the files in the `analysis` which are not in the the supplied  `args.files`? So it can't report 
`Specified file was not found in analysis`

Reply:
Fixed this line. Added checks for analysis files in supplied and vice versa.

https://github.com/icgc-argo/argo-data-submission/blob/5232a5bc7df766cc56cfa011b138e9c2a3b0b509/submission-receipt/main.py#L80-L85

Reply:
These cross checks may not be very necessary. Since the submission has completed, the purpose of this receipt is to let the user know what we have received. The idea of cross checks is good, however it will introduce some complexity in the overall workflow in providing the `List of files to verify against analysis ID` especially when the submitted files are CRAM.
How do you think? @edsu7

Reply:
Added a `skip-check` as a compromise. If unnecessary check can be skipped.

https://github.com/icgc-argo/argo-data-submission/blob/e0737f8113ccdeee015179b73af6cb4fe02ddb5e/submission-receipt/main.py#L89-L95